File: Damas C#/ProyectoDamasIng/ProyectoDamasIng/FichaNegra.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProyectoDamasIng
{
    class FichaNegra:Ficha
    {
        public FichaNegra(int fila, int columna):base("negro", fila, columna)
        {

        }
        //PosiblesMov se encarga de regresar un arreglo conteniendo los distintos espacios a los cuales la ficha se puede mover
        //el resultado arrojado varía segun la posición de la ficha
        override
        public List<Ficha> PosiblesMov()
        {
            this.posiblesMov = new List<Ficha>();
            
            if (this.fila != 7 && this.columna == 0)
            {
                this.posiblesMov.Add(new FichaNegra(this.fila + 1, this.columna + 1));
            }
            else if(this.fila != 7  && this.columna == 7)
            {
                this.posiblesMov.Add(new FichaNegra(this.fila + 1, this.columna - 1));
            }
            else if(this.fila != 7)
            {
                this.posiblesMov.Add(new FichaNegra(this.fila + 1, this.columna + 1));
                this.posiblesMov.Add(new FichaNegra(this.fila + 1, this.columna - 1));
            }

            return this.posiblesMov;
        }
        //PosiblesComer se encarga de regresar un arreglo conteniendo los distintos espacios a los cuales la ficha se puede mover para comer
        //el resultado arrojado varía segun la posición de la ficha
        override
        public List<Ficha> PosiblesComer()
        {
            this.posiblesComer = new List<Ficha>();
            if (this.fila < 6 && this.columna < 2)
            {
                this.posiblesComer.Add(new FichaNegra(this.fila + 2, this.columna + 2));
            }
            else if (this.fila < 6 && this.columna > 5)
            {
                this.posiblesComer.Add(new FichaNegra(this.fila + 2, this.columna - 2));
            }
            else if (this.fila < 6)
            {
                this.posiblesComer.Add(new FichaNegra(this.fila + 2, this.columna + 2));
                this.posiblesComer.Add(new FichaNegra(this.fila + 2, this.columna - 2));
            }

            return this.posiblesComer;
        }
    }
}

File: 02_DamasPython/02_DamasPython/FichaNegra.py
from Ficha import Ficha

class FichaNegra(Ficha):
    def __init__(self,fila,columna):
        super().__init__(fila,columna,"negra",'negra')
       
        self.__posiblesMov = []
        self.__posiblesComer = []
        self.__fila = fila
        self.__columna = columna
    
    @property
    def posiblesMov(self):
        
        return self.__posiblesMov
    @property   
    def posiblesComer(self):
        
        return self.__posiblesComer
    
    def PosiblesMov(self):
           
            if self.__fila != 7 and self.__columna == 0:
            
                self.__posiblesMov.append( FichaNegra(self.__fila + 1, self.__columna + 1))
             
            
            elif self.__fila != 7 and self.__columna == 7:
            
                self.__posiblesMov.append(FichaNegra(self.__fila + 1, self.__columna - 1))
              
            
            elif self.__fila != 7:
            
                self.__posiblesMov.append(FichaNegra(self.__fila + 1, self.__columna + 1))
                self.__posiblesMov.append( FichaNegra(self.__fila + 1, self.__columna - 1))
               
                    
            return self.__posiblesMov
    
     
    def PosiblesComer(self):
            
            if self.__fila <6  and self.__columna < 2:
            
                self.__posiblesComer.append( FichaNegra(self.__fila + 2, self.__columna + 2))
            
            elif self.__fila <6  and self.__columna > 5:
            
                self.__posiblesComer.append(FichaNegra(self.__fila + 2, self.__columna - 2))
            
            elif self.__fila <6:
            
                self.__posiblesComer.append(FichaNegra(self.__fila + 2, self.__columna + 2))
                self.__posiblesComer.append(FichaNegra(self.__fila + 2, self.__columna - 2))
            

            return self.__posiblesComer
        
    